ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -535,6 +537,91 @@ public void shouldCleanUpInternalTopicSchemasFromSchemaRegistry() throws Excepti
     )));
   }
 
+  @Test
+  public void shouldCleanUpInternalTopicsOnClose() {
+    // Given:
+    final QueryMetadata query = ksqlEngine.execute(
+        "select * from test1;",
+        KSQL_CONFIG, Collections.emptyMap()).get(0);
+
+    query.start();
+
+    // When:
+    query.close();
+
+    // Then:
+    verify(topicClient).deleteInternalTopics(query.getQueryApplicationId());
+  }
+
+  @Test
+  public void shouldNotCleanUpInternalTopicsOnCloseIfQueryNeverStarted() {
+    // Given:
+    final QueryMetadata query = ksqlEngine.execute(
+        "create stream s1 with (value_format = 'avro') as select * from test1;",
+        KSQL_CONFIG, Collections.emptyMap()).get(0);
+
+    // When:
+    query.close();
```

Comment:
Shouldn't this be a test in `QueryMetadataTest` instead of here? It tests `close()` method for `QueryMetadata`.

Reply:
See below.